mist-core/src/main/java/edu/snu/mist/core/task/groupaware/GroupAllocationTableModifierImpl.java

```diff
@@ -188,8 +188,10 @@ public void run() {
             }
             case GROUP_REMOVE: {
               final Group group = (Group) event.getValue();
-              removeGroupInWriterThread(group);
+              final ApplicationInfo applicationInfo = group.getApplicationInfo();
               groupMap.remove(group.getGroupId());
+              applicationInfo.removeGroup(group);
```

Comment:
Could you add comments on `removeGroup` and `addGroup` that these methods should be called into `GroupAllocationTableModifier`?

Reply:
Addressed in 25c0a01